hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -297,6 +297,30 @@ public void enableCacheOnWrite() {
     this.cacheBloomsOnWrite = true;
   }
 
+  /**
+   * If hbase.rs.cachecompactedblocksonwrite configuration is set to true and
+   * 'totalCompactedFilesSize' is lower than 'cacheCompactedDataOnWriteThreshold',
+   * enables cache on write for below properties:
+   * - cacheDataOnWrite
+   * - cacheIndexesOnWrite
+   * - cacheBloomsOnWrite
+   *
+   * Otherwise, sets 'cacheDataOnWrite' only to false.
+   *
+   * @param totalCompactedFilesSize the total size of compacted files.
+   * @return true if the checks mentioned above pass and the cache is enabled, false otherwise.
+   */
+  public boolean enableCacheOnWrite(long totalCompactedFilesSize) {
```

Comment:
Should we include something to distinguish that this is specifically for compactions in the method name since the CacheConfig is used in more places than just compactions?

Reply:
Renamed it on last commit.